superset-frontend/cypress-base/cypress/integration/dashboard/nativeFilters.test.ts

```diff
@@ -602,12 +332,314 @@ describe('Nativefilters Sanity test', () => {
+      .find(nativeFilters.filtersPanel.filterName)
+      .should('have.attr', 'value', testItems.topTenChart.filterColumn);
+  });
+
+  it('User can stop filtering when filter is removed', () => {
+    expandFilterOnLeftPanel();
+    enterNativeFilterEditModal();
+    fillNativeFilterForm(
+      testItems.filterType.value,
+      testItems.topTenChart.filterColumn,
+      testItems.datasetForNativeFilter,
+      testItems.topTenChart.filterColumn,
+    );
+    inputNativeFilterDefaultValue(testItems.filterDefaultValue);
+    saveNativeFilterSettings();
+    WORLD_HEALTH_CHARTS.forEach(waitForChartLoad);
+    cy.get(dataTestChartName(testItems.topTenChart.name)).within(() => {
+      cy.contains(testItems.filterDefaultValue).should('be.visible');
+      cy.contains(testItems.filterOtherCountry).should('not.exist');
+    });
+    cy.get(nativeFilters.filterItem)
+      .contains(testItems.filterDefaultValue)
+      .should('be.visible');
+    validateFilterNameOnDashboard(testItems.topTenChart.filterColumn);
+    enterNativeFilterEditModal();
+    deleteNativeFilter();
+    saveNativeFilterSettings();
+    WORLD_HEALTH_CHARTS.forEach(waitForChartLoad);
+    cy.get(dataTestChartName(testItems.topTenChart.name)).within(() => {
+      cy.contains(testItems.filterDefaultValue).should('be.visible');
+      cy.contains(testItems.filterOtherCountry).should('be.visible');
+    });
+  });
+
+  it('user can delete dependent filter', () => {
+    expandFilterOnLeftPanel();
+    enterNativeFilterEditModal();
+    fillNativeFilterForm(
+      testItems.filterType.value,
+      testItems.topTenChart.filterColumnRegion,
+      testItems.datasetForNativeFilter,
+      testItems.topTenChart.filterColumnRegion,
+    );
+    clickOnAddFilterInModal();
+    fillNativeFilterForm(
+      testItems.filterType.value,
+      testItems.topTenChart.filterColumn,
+      testItems.datasetForNativeFilter,
+      testItems.topTenChart.filterColumn,
+    );
+    cy.get(nativeFilters.filterConfigurationSections.displayedSection).within(
+      () => {
+        cy.contains('Values are dependent on other filters')
+          .should('be.visible')
+          .click();
+      },
+    );
+    addParentFilterWithValue(0, testItems.topTenChart.filterColumnRegion);
+    // remove year native filter to cause it disappears from parent filter input in global sales
+    cy.get(nativeFilters.modal.tabsList.removeTab)
+      .should('be.visible')
+      .first()
+      .click();
+    // make sure you are seeing global sales filter which had parent filter
+    cy.get(nativeFilters.modal.tabsList.filterItemsContainer)
+      .children()
+      .last()
+      .click();
+    //
+    cy.wait(1000);
+    cy.get(nativeFilters.filterConfigurationSections.displayedSection).within(
+      () => {
+        cy.contains('Values are dependent on other filters').should(
+          'not.exist',
+        );
+      },
+    );
+  });
+
+  it('User can create filter depend on 2 other filters', () => {
```

Comment:
We could have more functions to reduce the code even more like `fillRegionFilter`, `fillCountryCodeFilter`, `fillCountryNameFilter` and reuse them everywhere. You can still keep the original function for other combinations if needed.